packages/client/src/index.ts

```diff
@@ -164,54 +166,87 @@ export class SunriseStakeClient {
-        sunriseStakeState.gsolMint.toBuffer(),
-      ],
-      ASSOCIATED_TOKEN_PROGRAM_ID
-    )[0];
-    const [gsolMintAuthority] = findGSolMintAuthority(this.config);
-    this.msolTokenAccountAuthority = findMSolTokenAccountAuthority(
-      this.config
-    )[0];
-    this.bsolTokenAccountAuthority = findBSolTokenAccountAuthority(
-      this.config
-    )[0];
+    [this.marinade, this.marinadeState] = await this.makeMarinade();
+
+    this.blazeState = await this.makeBlaze();
+
+    this.lockClient = await LockClient.build(
+      this.config,
+      this.program,
+      this.staker
+    );
+  }
+
+  private async getTokenConfig(
+    config: SunriseStakeConfig
+  ): Promise<SunriseTokenConfig> {
+    const gsolMintAuthority = findGSolMintAuthority(config);
+
+    const msolTokenAccountAuthority = findMSolTokenAccountAuthority(config);
+    const msolTokenAccount = await utils.token.associatedAddress({
+      mint: this.env.marinade.msolMintAddress,
+      owner: msolTokenAccountAuthority[0],
+    });
+
+    const liqPoolTokenAccount = await utils.token.associatedAddress({
+      mint: this.env.marinade.lpMintAddress,
+      owner: msolTokenAccountAuthority[0],
+    });
+
+    const bsolTokenAccountAuthority = findBSolTokenAccountAuthority(config);
+    const bsolTokenAccount = await utils.token.associatedAddress({
+      mint: this.env.blaze.bsolMint,
+      owner: bsolTokenAccountAuthority[0],
+    });
+
+    return {
+      gsolMintAuthority,
+      msolTokenAccount,
+      msolTokenAccountAuthority,
+      liqPoolTokenAccount,
+      bsolTokenAccount,
+      bsolTokenAccountAuthority,
+    };
+  }
+
+  private async makeMarinade(): Promise<[Marinade, MarinadeState]> {
+    if (!this.config || !this.tokenConfig) throw new Error("Init not called");
 
     const marinadeConfig = new MarinadeConfig({
       connection: this.provider.connection,
       publicKey: this.provider.publicKey,
       proxyStateAddress: this.env.state,
-      proxySolMintAuthority: gsolMintAuthority,
+      proxySolMintAuthority: this.tokenConfig.gsolMintAuthority[0],
       proxySolMintAddress: this.config.gsolMint,
-      msolTokenAccountAuthority: this.msolTokenAccountAuthority,
+      msolTokenAccountAuthority: this.tokenConfig.msolTokenAccountAuthority[0],
       proxyTreasury: this.config.treasury,
     });
-    this.marinade = new Marinade(marinadeConfig);
-    this.marinadeState = await this.marinade.getMarinadeState();
-    this.msolTokenAccount = await utils.token.associatedAddress({
-      mint: this.marinadeState.mSolMintAddress,
-      owner: this.msolTokenAccountAuthority,
-    });
-    this.liqPoolTokenAccount = await utils.token.associatedAddress({
-      mint: this.marinadeState.lpMint.address,
-      owner: this.msolTokenAccountAuthority,
-    });
+
+    const marinade = new Marinade(marinadeConfig);
+    const marinadeState = await marinade.getMarinadeState();
+
+    return [marinade, marinadeState];
+  }
+
+  // TODO
```

Comment:
can you add some more detail here of what is remaining?

Reply:
That shouldn't be there. I'll remove it now